Source/items.h

```diff
@@ -165,6 +165,7 @@ enum icreateinfo_flag {
 enum icreateinfo_flag2 {
 	// clang-format off
 	CF_HELLFIRE = 1,
+	CF_NEW_UNIQ = 2,

```

Comment:
Some thoughts...

I like the idea of adding a flag for the new behavior. We could also think a little ahead,

Previous we had discussions about the crap... sub-optimal random generator of vanilla.
We also also have PR's open that replaces them (#2084 and #2261).
But both PR's don't touch item generators, cause this would break compatibility.

But with a similar flag (or the same?) like you did, we could change this.
With this we would stay compatible with vanilla (vanilla to devilutionX) but still create new (better) items.
The drawback would be that the new items would morph if loaded in vanilla (like the uniques).

This could be outside of this scope of this pr. I wanted to mention it as food for thoughts. And cause if we combine this we could save one flag. 😁

Reply:
yeah but I don't think it's the best time to mess around with all magic items :P uniques are enough for now + we got a whole uint32 to use, we don't have to save space for now : P